src/corecel/cont/detail/SpanImpl.hh

```diff
@@ -8,14 +8,39 @@
 #pragma once
 
 #include <cstddef>
+#include <type_traits>
 
 #include "corecel/Assert.hh"
 #include "corecel/Macros.hh"
+#include "corecel/OpaqueId.hh"
+#include "corecel/data/LdgIterator.hh"
 
 namespace celeritas
 {
 namespace detail
 {
+template<class T>
+struct SpanTraits
+{
+    using element_type = T;
+    using pointer = std::add_pointer_t<T>;
+    using const_pointer = std::add_pointer_t<T const>;
+    using iterator = pointer;
+    using const_iterator = const_pointer;
+    using reference = std::add_lvalue_reference_t<T>;
+    using const_reference = std::add_lvalue_reference_t<T const>;
+};
+template<class T>
+struct SpanTraits<LdgValue<T>>
+{
+    using element_type = typename LdgValue<T>::value_type;
+    using pointer = std::add_pointer_t<T const>;
+    using const_pointer = pointer;
+    using iterator = LdgIterator<T const>;
+    using const_iterator = iterator;
+    using reference = std::remove_const_t<T>;
+    using const_reference = std::remove_const_t<T>;
```

Comment:
Since all LDG data is returned by value, and `s_.data[0]` (I think?) returns an ldg-fetched value, shouldn't these both return `element_type`?

Reply:
`element_type` is const and we don't want to return a const value.